### PR TITLE
Issue #284 python 2 3 problems (No such file or directory: 'python' when installing python 3.5.2 after uninstalling python 2.7)

### PR DESCRIPTION
**Pull request by aaronki**
Fixing issue #284: No such file or directory: 'python' when installing python 3.5.2 after uninstalling python 2.7
by explicitly calling specific python version

### CODE REVIEW

File: examples/spear_qcp/scenario.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-algo = python -u ./target_algorithm/scripts/SATCSSCWrapper.py --mem-limit 1024 --script ./target_algorithm/spear-python/spearCSSCWrapper.py 
+algo = python2 -u ./target_algorithm/scripts/SATCSSCWrapper.py --mem-limit 1024 --script ./target_algorithm/spear-python/spearCSSCWrapper.py
```
Comment (mfeurer) [inline examples/spear_qcp/scenario.txt]: Is there a reason to specifically use `python2` here?

Comment (aaronki) [reply examples/spear_qcp/scenario.txt]: This prevents the python2 script SATCSSCWrapper.py from being run with python3 when python3 is the default, i.e. on Linux /usr/bin/python is a symlink to python3. As no features that are incompatible between python2 and python3 are used in the script, this original code magically works but in the general case it does not. This is the reason I explicitly call python2.

**Comment by aaronki** [comment]
@KEggensperger see my answer to mfeurer above

File: examples/spear_qcp/scenario.txt (hunk 1 above)
Comment (mfeurer) [reply examples/spear_qcp/scenario.txt]: If the script works in python3 as well, I'd rather change the script to `python3` to make us more forward-compatible.

Comment (aaronki) [reply examples/spear_qcp/scenario.txt]: This means I also have to change the shebang line in SATCSCWrapper.py. I'm doing this without explicitly reviewing the code. I consider this bad practice (the reason I changed it to explicitly use python2 and not python3) but for this case I'm doing it anyway.

Comment (mfeurer) [reply examples/spear_qcp/scenario.txt]: @mlindauer any issues here?

Comment (mlindauer) [reply examples/spear_qcp/scenario.txt]: There is no shebang in SATCSSCWrapper.py. So, I don't understand it.

in general, we don't support Python 2 and thus, all our examples should be based on Python 3

**Pull request merged**